LibPDF: Don't use unsanitized values in error messages

tcl3 opened the pull request:
Error messages may be formatted with non UTF-8 data, so `String` isn't suitable for this purpose.

An alternative approach would be to not use raw data from the input file in the error message, or to sanitize it before it is included.

This fixes oss fuzz issues: [62065](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=62065) and [57324](https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=57324)

tcl3 commented:
My latest push adds a regression test for this issue.

awesomekling commented:
> An alternative approach would be to not use raw data from the input file in the error message, or to sanitize it before it is included.

I think that would be the better approach, as otherwise the error strings will be much less useful to clients of LibPDF.

tcl3 commented:
I've updated the PR so that any input which may be invalid UTF-8 is no longer included in error messages. I've added conditional debug statements in these cases, so that these values can still be seen if needed.
